components/connectivity-adapter/internal/appregistry/service/converter.go

```diff
@@ -12,25 +21,420 @@ func NewConverter() *converter {
 	return &converter{}
 }
 
-func (c *converter) DetailsToGraphQLInput(in model.ServiceDetails) (graphql.ApplicationRegisterInput, error) {
-	// This is just a temporary fixture for testing purposes
-	// TODO: Replace with production-grade implementation
-	return graphql.ApplicationRegisterInput{
-		Name:           "wordpress",
-		ProviderName:   ptr.String("provider name"),
-		Description:    ptr.String("my first wordpress application"),
-		HealthCheckURL: ptr.String("http://mywordpress.com/health"),
-		Labels: &graphql.Labels{
-			"group":     []interface{}{"production", "experimental"},
-			"scenarios": []interface{}{"DEFAULT"},
-		},
-	}, nil
+func (c *converter) DetailsToGraphQLInput(deprecated model.ServiceDetails) (graphql.ApplicationRegisterInput, error) {
+
+	outLabels := graphql.Labels{}
+
+	out := graphql.ApplicationRegisterInput{
+		Name:         deprecated.Name,
+		Description:  ptrStringOrNilForEmpty(deprecated.Description),
+		ProviderName: ptrStringOrNilForEmpty(deprecated.Provider),
+	}
+
+	if deprecated.ShortDescription != "" {
+		outLabels[unmappedFieldShortDescription] = deprecated.ShortDescription
+	}
+	if deprecated.Identifier != "" {
+		outLabels[unmappedFieldIdentifier] = deprecated.Identifier
+	}
+	if deprecated.Api != nil {
+
+		outApi := &graphql.APIDefinitionInput{
+			TargetURL: deprecated.Api.TargetUrl,
+		}
+
+		if deprecated.Api.ApiType != "" {
+			if outApi.Spec == nil {
+				outApi.Spec = &graphql.APISpecInput{}
+			}
+			if strings.ToLower(deprecated.Api.ApiType) == "odata" {
+				outApi.Spec.Type = graphql.APISpecTypeOdata
+				outApi.Spec.Format = graphql.SpecFormatJSON // this field is required, so we arbitrary chose JSON
+			} else {
+				outApi.Spec.Type = graphql.APISpecTypeOpenAPI // quite brave assumption that it will be OpenAPI
```

Comment:
> quite brave assumption

![tenor-188709088](https://user-images.githubusercontent.com/7155799/73066113-7e6fbf00-3ea5-11ea-9ad9-85cfe934ef13.gif)

What about XML with ODATA? I'm not 100% sure, but I believe that App Registry saved the spec regardless the format and type, and then Headless CMS (Rafter) detected the type and uploaded proper spec. Please confirm it with corresponding teams.

If that was the case, unfortunately we need to do the same 🙁 Of course not in the same PR, but still, we need to detect type and format.